### PR TITLE
Add creator dockstore

### PR DESCRIPTION
**Pull request by lldelisle**
The goal is to propagate creator info from workflows to dockstore.yml.

Feel free to modify the code because I am not super happy about it.

### CODE REVIEW

File: planemo/workflow_lint.py [hunk 1]
```diff
@@ -60,21 +60,52 @@ class WorkflowLintContext(LintContext):
 
 def generate_dockstore_yaml(directory: str, publish: bool = True) -> str:
     workflows = []
-    for workflow_path in find_workflow_descriptions(directory):
+    all_workflow_paths = list(find_workflow_descriptions(directory))
+    for workflow_path in all_workflow_paths:
         test_parameter_path = f"{workflow_path.rsplit('.', 1)[0]}-tests.yml"
         workflow_entry: Dict[str, Any] = {
             # TODO: support CWL
+            "name": "main" if len(all_workflow_paths) == 1 else os.path.basename(workflow_path).replace(".ga$", ""),
```
Comment (mvdbeek) [inline planemo/workflow_lint.py:68]:
```suggestion
            "name": "main" if len(all_workflow_paths) == 1 else os.path.basename(workflow_path).split(".ga")[0],
```

File: planemo/workflow_lint.py [hunk 2]
```diff
@@ -60,21 +60,52 @@ class WorkflowLintContext(LintContext):
...
+                author = {}
+                # Put name first
+                if "name" in creator:
+                    author["name"] = creator["name"]
+                for field, value in creator.items():
+                    if field in ["class", "name"]:
+                        continue
+                    if field == "identifier":
+                        # Check if it is an orcid:
+                        orcid = re.findall(
```
Comment (mvdbeek) [inline planemo/workflow_lint.py:93]: There's also a more compact regex here: https://gist.github.com/asencis/644f174855899b873131c2cabcebeb87?permalink_comment_id=4210539#gistcomment-4210539

Should we check that the orcid is valid ?

`requests.get("https://orcid.org/0000-0002-1964-4960", headers={"Accept": "application/xml"}).status_code == 200` would tell us that.

Comment (lldelisle) [reply planemo/workflow_lint.py]: Great. I put it.

**Review by mvdbeek — APPROVED** [review]

**Comment by lldelisle** [comment]
@mvdbeek the test on dev local conda is still failing.

**Comment by mvdbeek** [comment]
I think @nsoranzo figured out what's up with that.

**Comment by mvdbeek** [comment]
Thanks, I'm excited to see this on dockstore!

**Comment by lldelisle** [comment]
Yes but that's why automerge did not work...
Thanks

**Comment by jmchilton** [comment]
Amazing, amazing, amazing! Thanks so much - this is really cool.

**Comment by nsoranzo** [comment]
> I think @nsoranzo figured out what's up with that.

I do have the fix for Galaxy dev, working on adding a test but got stuck in some test refactoring. Will open a PR later today, in a symposium at the moment.

**Pull request merged**